src/processor/operator/persistent/copy_to_parquet.cpp

```diff
@@ -10,17 +10,25 @@ void CopyToParquetLocalState::init(
     CopyToInfo* info, storage::MemoryManager* mm, ResultSet* resultSet) {
     auto copyToInfo = reinterpret_cast<CopyToParquetInfo*>(info);
     ft = std::make_unique<FactorizedTable>(mm, std::move(copyToInfo->tableSchema));
+    numTuplesInFT = 0;
+    countingVec = nullptr;
     vectorsToAppend.reserve(info->dataPoses.size());
     for (auto& pos : info->dataPoses) {
-        vectorsToAppend.push_back(resultSet->getValueVector(pos).get());
+        auto vec = resultSet->getValueVector(pos).get();
```

Comment:
Don't do this here because every vector you see here will be flat. Instead generate a countingVecPos at mapper and pass though constructor.